### ISSUE TITLE
Support XEP-0313 aka MAM (Message Archive Mgmt)

### ISSUE THREAD

**Issue by ghost** [psi-im/psi#62]
My initial wish is here: http://mail.jabber.org/pipermail/juser/2013-May/006962.html
I have no strict sight how it shoud look in UI.
I am ready to help actively with testing and debugging, and to provide MAM-enabled private jabber server for developer.

**Comment by rapgro** [comment]
duplication in psi-plus/main/issues/618

**Comment by Vitozz** [comment]
Please don't duplicate issues

**Comment by rapgro** [comment]
@Vitozz psi-plus is further development of psi, psi gets stable features only.

**Comment by ValdikSS** [comment]
I offer $1000 bounty for 0313 MAM support in Psi+. This should be implemented either directly in a code or as a plugin, and should be fully compatible with the specification and with existing server implementations. It should work with multiple connections on one account (multiple resources).

**Comment by NordMike** [comment]
@ValdikSS Is this bounty still valid?

**Comment by ValdikSS** [comment]
@NordMike Yes. As far as I know, MAM have been in development, but I'm not sure of the current state.

**Comment by ValdikSS** [comment]
Contact Psi/Psi+ development team first, to learn about status.

**Comment by NordMike** [comment]
@ValdikSS Where should I ask?

**Comment by ValdikSS** [comment]
psi-dev@conference.jabber.ru room.

**Comment by NordMike** [comment]
OK.

**Comment by Screemer1** [comment]
Hi, first post here. 

Any news on MAM for PSI+ ?
I really miss it from previously using Gajim and using Pix-Art messanger on my Android devices..
Perhaps it could be added as a plugin by someone if it cannot be added to core PSI+ ?

I got a reply on the forum from Boris. 
" This function is in plans of Psi development. And some necessary and important changes were already done. But unfortunately our developer who did most of work is not active in the project for now. "

**Comment by liuch** [comment]
Yes, I am doing this too slowly. I don't have enough time for this. I'm sorry. I will try to spend more time on it.

**Comment by Screemer1** [comment]
That would really be great  liuch, it would make PSI+ so much better as a client.

**Comment by Screemer1** [comment]
Any news? I've seen there has been a few updates in the personal release tree on Sourceforge. :)
Not that I want to be bitching but I really miss this function. :)
I've also found this in the Google group [here](https://groups.google.com/forum/#!topic/psi-users/ozKn8hc_yVA)

**Comment by cweiske** [comment]
Pull request/patch is at https://github.com/psi-im/iris/pull/59 (presumably dead)

**Comment by Neustradamus** [comment]
I think it will be nice to add this in the Google Summer of Code 2023:
- https://wiki.xmpp.org/web/Google_Summer_of_Code_2023

**Comment by Neustradamus** [comment]
@yatigautam, @butyoutried, @tech-bash: Maybe you could you continue the current PR?

Linked to: XEP-0313: Message Archive Management: MAM:
- https://xmpp.org/extensions/xep-0313.html
- https://github.com/psi-im/psi/issues/62
- https://github.com/psi-plus/main/issues/618
- https://github.com/psi-im/iris/pull/59

**Comment by mcneb10** [comment]
My MAM PR was just merged into iris!

https://github.com/psi-im/iris/pull/73

**Comment by Neustradamus** [comment]
@Ri0n: @mcneb10 has done an important work for MAM, can you look to have a perfect XMPP Client?

Note: A new XEP-0313 version has been released - 1.1.2 (2024-09-18):
- https://xmpp.org/extensions/xep-0313.html
- https://services.w3.org/htmldiff?doc1=https://xmpp.org/extensions/attic/xep-0313-1.1.1.html&doc2=https://xmpp.org/extensions/attic/xep-0313-1.1.2.html

It is good?

**Comment by Neustradamus** [comment]
Since the good work done by @mcneb10:
- https://github.com/psi-im/iris/pull/73

And after XEP-0313 1.1.2 (2024-09-18), a new XEP-0313 version has been released - 1.1.3 (2025-04-09):
- https://xmpp.org/extensions/xep-0313.html
- https://services.w3.org/htmldiff?doc1=https://xmpp.org/extensions/attic/xep-0313-1.1.2.html&doc2=https://xmpp.org/extensions/attic/xep-0313-1.1.3.html
- https://services.w3.org/htmldiff?doc1=https://xmpp.org/extensions/attic/xep-0313-1.1.1.html&doc2=https://xmpp.org/extensions/attic/xep-0313-1.1.2.html

**Comment by kayodde** [comment]
Any updates regarding MAM support into psi+?
Atm, I am using a hacked together patch with the iris backend to try and get something working. However my programming knowledge is nowhere near extensive enough to properly implement anything...
